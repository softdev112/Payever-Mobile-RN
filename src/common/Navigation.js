/* eslint no-prototype-builtins: 0, no-continue: 0 */

import { PropTypes } from 'react';
import { Provider } from 'mobx-react/native';
import { Navigation, Navigator } from 'react-native-navigation';
import Screen from 'react-native-navigation/src/Screen';

import type Store from '../store';

const SPECIAL_REACT_KEYS = { children: true, key: true, ref: true };

patchRnnScreenClass(Screen);


class MobxRnnProvider extends Provider {
  props: {
    store: Object;
  };

  context: {
    mobxStores: Object;
  };

  getChildContext() {
    const stores = {};

    // inherit stores
    const baseStores = this.context.mobxStores;
    if (baseStores) {
      /* eslint guard-for-in: 0 */
      for (const key in baseStores) {
        stores[key] = baseStores[key];
      }
    }

    // add own stores
    for (const key in this.props.store) {
      if (!SPECIAL_REACT_KEYS[key]) {
        stores[key] = this.props.store[key];
      }
    }

    return { mobxStores: stores };
  }
}

export function registerScreens(screens: Object, store: Store) {
  for (const i in screens) {
    if (!screens.hasOwnProperty(i)) continue;
    Navigation.registerComponent(i, () => screens[i], store, MobxRnnProvider);
  }
}

export function showScreen(screenId) {
  Navigation.startSingleScreenApp({
    screen: { screen: screenId },
    appStyle: {
      screenBackgroundColor: '#ffffff',
    },
    drawer: {
      right: {
        screen: 'core.SideMenu',
      },
      style: {
        drawerShadow: 'NO',
      },
      disableOpenGesture: true,
    },
  });
}

export function toggleMenu(navigator: Navigator, params = {}) {
  navigator.push({
    screen: 'core.SideMenu',
    animated: false,
    ...params,
  });
}

export function hideMenu(navigator: Navigator, params = {}) {
  navigator.pop({
    animated: false,
    ...params,
  });
}

/**
 * Patch RNN Screen class to pass navigator to childContext
 */
function patchRnnScreenClass(ScreenClass: Screen) {
  //noinspection JSUndefinedPropertyAssignment
  ScreenClass.childContextTypes = { navigator: PropTypes.object };
  ScreenClass.prototype.getChildContext = function getChildContext() {
    return { navigator: this.navigator };
  };
}