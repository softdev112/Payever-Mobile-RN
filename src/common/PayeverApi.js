import type AuthApi from '../modules/auth/api/AuthApi';
import type UserApi from '../modules/user/api/UserApi';

import { each, merge, isDate, now } from 'lodash';

type PayeverApiConfig = {
  baseUrl: string,
  clientId: string,
  clientSecret: string,
  accessToken: string,
  expiresIn: Date,
  refreshToken: string;
};

declare class PayeverResponse extends Response {
  data: {
    error?: string,
    error_description?: string
  };
  json(): Object;
}

export default class PayeverApi {
  auth: AuthApi;
  user: UserApi;

  baseUrl: string;
  clientId: string;
  clientSecret: string;

  accessToken: string;
  expiresIn: Date;
  refreshToken: string;

  constructor(config: PayeverApiConfig, childClasses: {[id: string]: Class}) {
    this.setConfig(config);

    each(childClasses, (Class, name) => {
      this[name] = new Class(this);
    });
  }

  setConfig(config: PayeverApiConfig) {
    if (typeof config.baseUrl === 'string' && config.baseUrl.endsWith('/')) {
      config.baseUrl = config.baseUrl.slice(0, -1);
    }

    const expires = config.expiresIn;
    if (expires && !isDate(expires) && isFinite(expires)) {
      config.expiresIn = new Date(now() + (expires - 10) * 1000);
    }

    merge(this, config);
  }

  async get(url: string, query: Object = null): Promise<Response> {
    query = {
      ...query,
      access_token: await this.getAccessToken()
    };
    return this.fetch(url, { query });
  }

  async fetch(url: string, options: Object = {}): Promise<PayeverResponse> {
    options.method = options.method || 'GET';
    const response: PayeverResponse = await fetch(
      this.normalizeUrl(url, options.query),
      options
    );
    const text = await response.text();
    try {
      response.data = JSON.parse(text);
    } catch(e) {
      console.log('PayeverApi: Error parsing JSON', text);
      response.data = {
        error: 'json_error',
        error_description: 'Wrong server response'
      }
    }

    return response;
  }

  async getAccessToken() {
    if (this.accessToken && this.expiresIn > new Date()) {
      return this.accessToken;
    }
    if (!this.refreshToken) {
      throw new Error('PayeverApi: refreshToken is null')
    }
    return await this.auth.refreshToken(this.refreshToken);
  }

  normalizeUrl(url: string, query: Object = null) {
    let fullUrl = this.baseUrl + url;
    if (query && Object.keys(query).length) {
      fullUrl += '?' + objectToQueryString(query)
    }
    return fullUrl;
  }
}

function objectToQueryString(data: Object): string {
  return Object.keys(data).map((key) => {
    return encodeURIComponent(key) + '=' + encodeURIComponent(data[key])
  }).join('&');
}


