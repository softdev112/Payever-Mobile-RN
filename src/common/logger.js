import { NativeModules } from 'react-native';
const StethoLogger = NativeModules.StethoLogger;

/**
 * It's just an experiment.
 * @param message
 */
global.log = function(message) {
  StethoLogger.sendChromeNotification('Console.messageAdded', {
    message: {
      source: 'other',
      level: 'log',
      type: 'log',
      text: { id : 1 },
      timestamp: Date.now(),
      line: 2,
      column: 13,
      url: 'file:///c:/page/script.js',
      executionContextId: 2,
      parameters: [
        { type: 'string', value: '{"test":1}' }
      ]
    }
  });

  StethoLogger.sendChromeNotification('Runtime.evaluate', {
    expression: 'console.log(123)'
  });
};