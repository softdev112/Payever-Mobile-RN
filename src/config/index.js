import generatedConfig from './_autogenerated';

export class Config {
  siteUrl: string;
  api: {
    baseUrl: string;
    clientId: string;
    clientSecret: string;
  };
  debug: {
    consoleLevel: boolean;
    logApiCall: boolean;
    logMobx: boolean;
    logSLevel: boolean;
    logSUrl: string;
    logWampCall: boolean;
  };

  constructor(data) {
    data.debug = data.debug || {};
    Object.assign(this, data);
  }
}

export default new Config(generatedConfig);