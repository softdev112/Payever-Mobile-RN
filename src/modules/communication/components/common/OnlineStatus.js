import { StyleSheet, View } from 'ui';

export default function OnlineStatus({ isOnline, size, style }: PropTypes) {
  const statusStyles = [styles.container, style];
  if (isOnline) {
    statusStyles.push(styles.onlineStatus);
  } else {
    statusStyles.push(styles.offlineStatus);
  }

  if (size !== undefined) {
    statusStyles.push({
      height: size,
      width: size,
      borderRadius: size / 2,
    });
  }

  return (
    <View style={statusStyles} />
  );
}

const styles = StyleSheet.create({
  container: {
    borderColor: '#d8d8d8',
    borderRadius: 4,
    borderWidth: 1,
    height: 8,
    width: 8,
  },

  offlineStatus: {
    backgroundColor: '#FFF',
    borderWidth: 1,
  },

  onlineStatus: {
    backgroundColor: '#75b636',
    borderWidth: 0,
  },
});

type PropTypes = {
  online: boolean;
};