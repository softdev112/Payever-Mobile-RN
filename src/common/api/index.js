import { merge } from 'lodash';
import { log } from 'utils';

import type AuthStore from '../../store/AuthStore';
import AuthApi from './AuthApi';
import BusinessApi from './BusinessApi';
import DashboardApi from './DashboardApi';
import UserApi from './UserApi';
import ProfilesApi from './ProfilesApi';
import MenuApi from './MenuApi';
import { showScreen } from '../Navigation';

export default class PayeverApi {
  auth: AuthApi;
  business: BusinessApi;
  dashboard: DashboardApi;
  user: UserApi;
  profiles: ProfilesApi;
  menu: MenuApi;

  baseUrl: string;
  clientId: string;
  clientSecret: string;

  authStore: AuthStore;

  constructor(config: PayeverApiConfig) {
    this.setConfig(config);
    this.registerSubApi();
  }

  registerSubApi() {
    this.auth      = new AuthApi(this);
    this.dashboard = new DashboardApi(this);
    this.business  = new BusinessApi(this);
    this.menu      = new MenuApi(this);
    this.user      = new UserApi(this);
    this.profiles  = new ProfilesApi(this);
  }

  setConfig(config: PayeverApiConfig) {
    if (typeof config.baseUrl === 'string' && config.baseUrl.endsWith('/')) {
      config.baseUrl = config.baseUrl.slice(0, -1);
    }

    merge(this, config);
  }

  async get(url: string, query: Object = null): Promise<ApiResp> {
    query = {
      ...query,
      access_token: await this.getAccessToken(),
    };
    return this.fetch(url, { query });
  }

  async post(url: string, formData: Object = null): Promise<ApiResp> {
    const options = {
      method: 'POST',
    };

    if (formData) {
      options.body = objectToPhpFormData(formData);
    }

    return this.fetch(url, options);
  }

  //noinspection ReservedWordAsName
  async delete(url: string, formData: Object = null): Promise<ApiResp> {
    const options = {
      method: 'DELETE',
    };

    if (formData) {
      options.body = objectToPhpFormData(formData);
    }

    return this.fetch(url, options);
  }

  //noinspection InfiniteRecursionJS
  async fetch(url: string, options: Object = {}): Promise<ApiResp> {
    options.method = options.method || 'GET';
    url = this.normalizeUrl(url, options.query);

    if (__DEV__) {
      log.debug(`${options.method} ${url}`);
    }

    const response: ApiResp = await fetch(url, options);
    const text = await response.text();
    try {
      response.data = JSON.parse(text);
      if (response.data && response.data.error) {
        response.error = response.data.error;
        response.errorDescription = response.data.errorDescription;
      }
    } catch (e) {
      response.data = {};
      response.error = 'json_error';
      response.errorDescription = 'Wrong server response';
    }

    if (!options.preventTokenRefresh &&
      response.error === 'invalid_grant') {
      const token = await this.auth.refreshToken(this.authStore.refreshToken);
      if (token) {
        return await this.fetch(url, { ...options, preventTokenRefresh: true });
      }
      showScreen('auth.Login');
    }

    if (__DEV__) {
      log.debug('Response data ', response.data);
    }

    return response;
  }

  async getAccessToken() {
    if (this.authStore.accessToken && this.authStore.expiresIn > new Date()) {
      return this.authStore.accessToken;
    }
    if (!this.authStore.refreshToken) {
      throw new Error('PayeverApi: refreshToken is null');
    }
    return await this.auth.refreshToken(this.authStore.refreshToken);
  }

  normalizeUrl(url: string, query: Object = null) {
    let fullUrl = this.baseUrl + url;
    if (query && Object.keys(query).length) {
      fullUrl += '?' + objectToQueryString(query);
    }
    return fullUrl;
  }
}

function objectToQueryString(data: Object): string {
  return Object.keys(data).map((key) => {
    return encodeURIComponent(key) + '=' + encodeURIComponent(data[key]);
  }).join('&');
}

function objectToPhpFormData(data: ObjectToFromData) {
  // Good example: POST /api/rest/v1/channel-subscription/{id}/create-store
  const formData = new FormData();

  if (!data || data.key === '' || !data.requestData) return formData;

  const typeDetector = Object.prototype.toString;
  const mainKey = data.key;
  const requestData = data.requestData;
  const keys = Object.keys(data);

  function appendValueToFormData(currentName, value) {
    switch (typeDetector.call(value)) {
      case '[object Object]': {
        const objKeys = Object.keys(value);
        objKeys.forEach(key =>
          appendValueToFormData(`${currentName}[${key}]`, value[key]));
      }
        break;

      case '[object Array]':
        value.forEach(element =>
          appendValueToFormData(`${currentName}[]`, element));
        break;

      default:
        formData.append(currentName, value);
        break;
    }
  }

  try {
    keys.forEach(key => {
      appendValueToFormData(`${mainKey}[${key}]`, requestData[key]);
    });
  } catch (e) {
    log.error(e.message);
  }

  return formData;
}

type ObjectToFromData = {
  key: string;
  requestData: Object;
};

type PayeverApiConfig = {
  baseUrl: string;
  clientId: string;
  clientSecret: string;
  accessToken: string;
  expiresIn: Date;
  refreshToken: string;
};