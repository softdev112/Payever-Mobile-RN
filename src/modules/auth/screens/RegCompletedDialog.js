import { Component, PropTypes } from 'react';
import { inject, observer } from 'mobx-react/native';
import type { Navigator } from 'react-native-navigation';
import { Icon, TextButton, StyleSheet, Text, View } from 'ui';

import { showScreen } from '../../../common/Navigation';
import type AuthStore from '../../../store/auth';

/* eslint-disable max-len */
const successMessage = `  New user was successfully created. Confirmation email was sent to your address.
                          Check out it please and confirm it. After that you will be able to login with
                          your email and password. Thanks and welcome to Payever!`.replace(/\s+/g, ' ');

const errorMessage = 'Sorry. There are some errors occurred. Try again later please!';
/* eslint-enable max-len */

@inject('auth')
@observer
export default class RegCompletedDialog extends Component {
  static contextTypes = {
    navigator: PropTypes.object.isRequired,
  };

  props: {
    auth: AuthStore;
  };

  context: {
    navigator: Navigator;
  };

  onCLoseDialog() {
    const { navigator } = this.context;
    navigator.dismissLightBox({ animated: true });
    showScreen('core.LaunchScreen');
  }

  render() {
    const { auth } = this.props;
    const isError = auth.error !== '';

    const iconStyle = [styles.icon];
    if (isError) {
      iconStyle.push(styles.errorIcon);
    }

    return (
      <View style={styles.container}>
        <Icon
          style={iconStyle}
          source={isError ? 'icon-alert-32' : 'icon-checked-64'}
        />

        { isError ? (
          <View>
            <Text style={styles.message}>{errorMessage}</Text>
            <Text style={styles.message}>{auth.error}</Text>
          </View>
        ) : (
          <Text style={styles.message}>{successMessage}</Text>
        )}

        <TextButton
          style={styles.btn}
          title="OK"
          onPress={::this.onCLoseDialog}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    borderRadius: 15,
    backgroundColor: '#FFF',
    alignItems: 'center',
    height: 300,
    width: 250,
    paddingHorizontal: 15,
    paddingTop: 15,
  },

  icon: {
    fontSize: 28,
    color: '$pe_color_green',
    marginBottom: 10,
  },

  errorIcon: {
    color: '$pe_color_red',
  },

  message: {
    fontSize: 15,
    fontFamily: '$font_family',
    textAlign: 'justify',
  },

  btn: {
    position: 'absolute',
    bottom: 20,
    right: 40,
  },
});