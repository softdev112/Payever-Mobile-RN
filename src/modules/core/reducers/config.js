import { Map } from 'immutable';
import config from '../../../config';

const initialState = Map(config);

/**
 * Share config through store
 */
export default function configReducer(state = initialState) {
  return state;
}