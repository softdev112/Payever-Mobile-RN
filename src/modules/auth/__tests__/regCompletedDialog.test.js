/* eslint-disable max-len */
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import RegCompletedDialog from '../screens/RegCompletedDialog';
import Store from '../../../store';
import config from '../../../config';
import navigator from '../../../../__mocks__/navigator';
import { showScreen } from '../../../common/Navigation';

jest.mock('../../../common/Navigation');

describe('modules/auth/screens/RegCompletedDialog', () => {
  let store;
  let auth;

  beforeEach(() => {
    store = new Store(config);
    auth = store.auth;
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('RegCompletedDialog should render correctly if auth.error does not set', () => {
    const tree = renderer.create(
      <RegCompletedDialog
        auth={auth}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('RegCompletedDialog should render correctly if auth.error is set', () => {
    auth.error = 'Test eror';

    const tree = renderer.create(
      <RegCompletedDialog
        auth={auth}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('RegCompletedDialog/onCLoseDialog() should call navigator.dismissLightBox and showScreen => core.LaunchScreen', () => {
    const wrapper = shallow(
      <RegCompletedDialog
        auth={auth}
      />,
      { context: { navigator } }
    );

    wrapper.dive()
      .find('TextButton')
      .at(0)
      .props()
      .onPress();

    expect(navigator.dismissLightBox).toHaveBeenCalled();
    expect(navigator.dismissLightBox).toHaveBeenCalledWith({ animated: true });
    expect(showScreen).toHaveBeenCalled();
    expect(showScreen).toHaveBeenCalledWith('core.LaunchScreen');
  });
});