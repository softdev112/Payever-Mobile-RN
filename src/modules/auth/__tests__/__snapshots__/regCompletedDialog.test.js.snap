// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules/auth/screens/RegCompletedDialog RegCompletedDialog should render correctly if auth.error does not set 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "borderRadius": 15,
      "flex": 1,
      "height": 300,
      "paddingHorizontal": 15,
      "paddingTop": 15,
      "width": 250,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    source="icon-checked-64"
    style={
      Array [
        Object {
          "color": "#75b636",
          "fontFamily": "payeverIcons",
          "fontSize": 64,
        },
        Array [
          Object {
            "color": "#75b636",
            "fontFamily": "payeverIcons",
            "fontSize": 64,
          },
          Array [
            Object {
              "color": "#75b636",
              "fontSize": 28,
              "marginBottom": 10,
            },
          ],
        ],
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#3d3d3d",
          "fontFamily": "Helvetica Neue",
          "fontWeight": "200",
        },
        Object {
          "fontFamily": "Helvetica Neue",
          "fontSize": 15,
          "textAlign": "justify",
        },
      ]
    }
  >
     New user was successfully created. Confirmation email was sent to your address. Check out it please and confirm it. After that you will be able to login with your email and password. Thanks and welcome to Payever!
  </Text>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
        },
        Object {
          "bottom": 20,
          "position": "absolute",
          "right": 40,
        },
      ]
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0084ff",
              "fontSize": 16,
              "fontWeight": "400",
            },
            undefined,
          ]
        }
      >
        OK
      </Text>
    </View>
  </View>
</View>
`;

exports[`modules/auth/screens/RegCompletedDialog RegCompletedDialog should render correctly if auth.error is set 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "borderRadius": 15,
      "flex": 1,
      "height": 300,
      "paddingHorizontal": 15,
      "paddingTop": 15,
      "width": 250,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    source="icon-alert-32"
    style={
      Array [
        Object {
          "fontFamily": "payeverIcons",
          "fontSize": 32,
        },
        Array [
          Object {
            "fontFamily": "payeverIcons",
            "fontSize": 32,
          },
          Array [
            Object {
              "color": "#75b636",
              "fontSize": 28,
              "marginBottom": 10,
            },
            Object {
              "color": "#ff3c30",
            },
          ],
        ],
      ]
    }
  >
    
  </Text>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#3d3d3d",
            "fontFamily": "Helvetica Neue",
            "fontWeight": "200",
          },
          Object {
            "fontFamily": "Helvetica Neue",
            "fontSize": 15,
            "textAlign": "justify",
          },
        ]
      }
    >
      Sorry. There are some errors occurred. Try again later please!
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#3d3d3d",
            "fontFamily": "Helvetica Neue",
            "fontWeight": "200",
          },
          Object {
            "fontFamily": "Helvetica Neue",
            "fontSize": 15,
            "textAlign": "justify",
          },
        ]
      }
    >
      Test eror
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
        },
        Object {
          "bottom": 20,
          "position": "absolute",
          "right": 40,
        },
      ]
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0084ff",
              "fontSize": 16,
              "fontWeight": "400",
            },
            undefined,
          ]
        }
      >
        OK
      </Text>
    </View>
  </View>
</View>
`;
